src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java

```diff
@@ -344,977 +374,702 @@ public sealed interface Subtask<T> extends Supplier<T> permits SubtaskImpl {
+         * return a result of the same type. Joiners returned by {@link
+         * #awaitAllSuccessfulOrThrow()} are suited to cases where the subtasks return
+         * results of different types.
+         *
+         * @param <T> the result type of subtasks
+         */
+        static <T> Joiner<T, Stream<Subtask<T>>> allSuccessfulOrThrow() {
+            return new Joiners.AllSuccessful<>();
+        }
+
+        /**
+         * {@return a new Joiner object that yields the result of any subtask that
+         * completed successfully}
+         * The {@code Joiner} causes {@code join} to throw if all subtasks fail.
+         *
+         * <p> The joiner's {@link Joiner#result()} method returns the result of a subtask
+         * that completed successfully. If all subtasks fail then the {@code result} method
+         * throws the exception from one of the failed subtasks. The {@code result} method
+         * throws {@code NoSuchElementException} if no subtasks were forked.
+         *
+         * @param <T> the result type of subtasks
+         */
+        static <T> Joiner<T, T> anySuccessfulResultOrThrow() {
+            return new Joiners.AnySuccessful<>();
+        }
+
+        /**
+         * {@return a new Joiner object that waits for subtasks to complete successfully}
+         * The {@code Joiner} <a href="StructuredTaskScope.html#Cancallation">cancels</a>
+         * the scope and causes {@code join} to throw if any subtask fails.
+         *
+         * <p> The joiner's {@link Joiner#result() result} method returns {@code null}
+         * if all subtasks complete successfully, or throws the exception from the first
+         * subtask to fail.
+         *
+         * @apiNote Joiners returned by this method are suited to cases where subtasks
+         * return results of different types. Joiners returned by {@link #allSuccessfulOrThrow()}
+         * are suited to cases where the subtasks return a result of the same type.
+         *
+         * @param <T> the result type of subtasks
+         */
+        static <T> Joiner<T, Void> awaitAllSuccessfulOrThrow() {
+            return new Joiners.AwaitSuccessful<>();
+        }
+
+        /**
+         * {@return a new Joiner object that waits for all subtasks to complete}
+         * The {@code Joiner} does not cancel the scope if a subtask fails.
+         *
+         * <p> The joiner's {@link Joiner#result() result} method returns {@code null}.
+         *
+         * @apiNote This Joiner is useful for cases where subtasks make use of
+         * <em>side-effects</em> rather than return results or fail with exceptions.
+         * The {@link #fork(Runnable) fork(Runnable)} method can be used to fork subtasks
+         * that do not return a result.
+         *
+         * <p> This Joiner can also be used for <em>fan-in</em> scenarios where subtasks
+         * for forked to handle incoming connections and the number of subtasks is unbounded.
```

Comment:
Sentence does not read well. Should 'for' have been 'are'?

```suggestion
         * are forked to handle incoming connections and the number of subtasks is unbounded.
```